pypeit/scripts/sensfunc.py

```diff
@@ -118,74 +128,88 @@ def main(args):
+
+                # Construct a primary FITS header that includes the spectrograph's
+                #   config keys for inclusion in the output sensfunc file
+                primary_hdr = io.initialize_header()
+                add_keys = (
+                    ['PYP_SPEC', 'DATE-OBS', 'TELESCOP', 'INSTRUME', 'DETECTOR']
+                    + spectrograph.configuration_keys() + spectrograph.raw_header_cards()
+                )
+                for key in add_keys:
+                    if key.upper() in hdul[0].header.keys():
+                        primary_hdr[key.upper()] = hdul[0].header[key.upper()]
+
+            # If the .sens file was passed in read it and overwrite default parameters
+            if args.sens_file is not None:
+                sensFile = inputfiles.SensFile.from_file(args.sens_file)
+                # Read sens file
+                par = pypeitpar.PypeItPar.from_cfg_lines(
+                            cfg_lines=spectrograph_config_par.to_config(),
+                            merge_with=(sensFile.cfg_lines,))
+            else:
+                par = pypeitpar.PypeItPar.from_cfg_lines(cfg_lines=spectrograph_config_par.to_config())
+
+            # If algorithm was provided override defaults. Note this does undo .sens
+            # file since they cannot both be passed
+            if args.algorithm is not None:
+                par['sensfunc']['algorithm'] = args.algorithm
+
+            # If flatfile was provided override defaults. Note this does undo .sens
+            # file since they cannot both be passed
+            if args.flatfile is not None:
+                par['sensfunc']['flatfile'] = args.flatfile
+
+            # If multi was set override defaults. Note this does undo .sens file
+            # since they cannot both be passed
+            if args.multi is not None:
+                # parse
+                multi_spec_det  = [int(item) for item in args.multi.split(',')]
+                par['sensfunc']['multi_spec_det'] = multi_spec_det
+
+            # TODO Add parsing of detectors here. If detectors passed from the
+            # command line, overwrite the parset values read in from the .sens file
+
+            # Instantiate the relevant class for the requested algorithm
+            sensobj = sensfunc.SensFunc.get_instance(file, ofile, par['sensfunc'],
+                                                     par_fluxcalib=par['fluxcalib'],
+                                                     debug=args.debug,
+                                                     chk_version=par['rdx']['chk_version'])
+            # Generate the sensfunc
+            sensobj.run()
+            msgs.info(f'Saved std FWHM as: {sensobj.spat_fwhm_std}')
+            # Write it out to a file, including the new primary FITS header
+            sensobj.to_file(ofile, primary_hdr=primary_hdr, overwrite=True)
+            msgs.info('-'*50)
+            msgs.info('-'*50)
+
+            # Write the par to disk
```

Comment:
@Adolfo1519 I just noticed that you commented out the lines below, resulting in not writing the parameters to a file, therefore the `--par_outfile` is not even used. Could you fix that? We definitely want to save those parameters to a file.

Reply:
Sorry, I have uncommented them. I do not recall why I would have commented those lines out.